connector/gitlab/gitlab.go

```diff
@@ -272,22 +269,31 @@ func (c *gitlabConnector) groups(ctx context.Context, client *http.Client) ([]st
 
 		link := resp.Header.Get("Link")
 
-		if len(reLast.FindStringSubmatch(link)) > 1 {
-			lastPageURL := reLast.FindStringSubmatch(link)[1]
-
-			if apiURL == lastPageURL {
-				break
-			}
-		} else {
+		apiURL = nextURL(apiURL, link)
+		if apiURL == "" {
 			break
 		}
+	}
+	return groups, nil
+}
 
-		if len(reNext.FindStringSubmatch(link)) > 1 {
-			apiURL = reNext.FindStringSubmatch(link)[1]
-		} else {
-			break
+func nextURL(url string, link string) string {
+	reNext := regexp.MustCompile("<([^>]+)>; rel=\"next\"")
```

Comment:
nit: move these to global vars so we don't re-compile them every time the function is called

```
var (
    reNext = regexp.MustCompile("...")
    reLast = regexp.MustCompile("...")
)

func nextURL() {
    // .. use reNext and reLast
}
```

Reply:
Will do!